[quant][graphmode][fx] Add support for fp16 bmm pattern

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52865 [quant][graphmode][fx][fp16] Add fp16 support for silu
* #52864 [quant][graphmode][fx][fp16] Add fp16 support for tanh
* #52863 [quant][graphmode][fx][fp16] Add fp16 support for sigmoid
* #52862 [quant][graphmode][fx][fp16] Add fp16 support for layer_norm
* #52811 [quant][graphmode][fx][fp16] Add fp16 support for torch.sum
* #52810 [quant][graphmode][fx][fp16] Add fp16 support for div
* #52809 [quant][graphmode][fx][fp16] Add fp16 support for sub
* **#52808 [quant][graphmode][fx] Add support for fp16 bmm pattern**

Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

jerryzh168 committed:
[quant][graphmode][fx] Add support for fp16 bmm pattern

Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on test/quantization/test_quantize_fx.py:
```diff
@@ -104,7 +104,7 @@ class BinaryOpNonQuantizedInput(torch.nn.Module):
         super().__init__()
         self.is_scalar = is_scalar
-        self.op = ibinary_op if is_inplace else binary_op
+        self.op = ibinary_op if ibinary_op and is_inplace else binary_op
```
optional: what's the difference between binary_op or ibinary_op?  Would be great if there was a comment

jerryzh168 replied:
oh that means inplace binary op

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -109,9 +110,12 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
-        self.qop = qbin_relu_op_mapping[self.bop] \
-            if self.relu_node is not None \
-            else qbin_op_mapping[self.bop]  # type: ignore
+        if self.bop in qbin_op_mapping:
```
optional nit (probably for an earlier PR): `bop` is hard to think about, maybe `binary_op`?  same for `qop`

jerryzh168 replied:
sure

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for fp16 bmm pattern"


Summary:
Add support for producing fp16 bmm pattern

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_bmm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26655616](https://our.internmc.facebook.com/intern/diff/D26655616)

[ghstack-poisoned]
